pde_superresolution/weno.py

```diff
@@ -143,13 +139,12 @@ def calculate_time_derivative(self, u: np.ndarray) -> np.ndarray:
     time_deriv = (flux - np.roll(flux, 1)) / self.dx
     return time_deriv
 
-  def reconstruction(
-      self,
-      u: np.ndarray,
-      p: np.ndarray,
-      s: np.ndarray,
-      d: np.ndarray
-  ) -> np.ndarray:
+  def u_plus_minus(self, u) -> Tuple[np.ndarray, np.ndarray]:
+      u_minus = self.reconstruction(np.roll(u, -1))
+      u_plus = self.reconstruction(u[::-1])[::-1]
```

Comment:
Let's discuss. I believe that this restores symmetry, but does it do upwinding properly? I think even the coefficients for `u_minus` need to change if the wind is blowing from the right instead of from the left.

It would be nice to write a test that verifies that we recover the "optimal" 5th order stencil (one point upwinded, depending on the direction of the velocity) for entirely smooth functions.

Reply:
> I think even the coefficients for `u_minus` need to change if the wind is blowing from the right instead of from the left.

I don't think so. As far as I understand the WENO reconstruction step is agnostic to the direction of the wind. Note that the development of the WENO formulae does not involve the physical equation, and the wind is a property of the equation, not of the solution (in the general case it is the characteristic, which is a direction determined by the Jacobian of the Differential operator). 

> It would be nice to write a test that verifies that we recover the "optimal" 5th order stencil (one point upwinded, depending on the direction of the velocity) for entirely smooth functions.

Agreed

Reply:
> I the wind is a property of the equation, not of the solution

That is, it of course depends on the local solution, but the mapping of the local solution to the characteristic direction is equation dependent and knowing only the solution, as we do when doing WENO reconstruction, there's no way of knowing where the wind is blowing to